Orm/Xtensive.Orm.SqlServer/Sql.Drivers.SqlServer/v11/Extractor.cs

```diff
@@ -46,12 +49,17 @@ ORDER BY
           var currentSchema = GetSchema(reader.GetInt32(0));
           var sequence = currentSchema.CreateSequence(reader.GetString(1));
           var descriptor = sequence.SequenceDescriptor;
-          descriptor.StartValue = reader.GetInt64(2);
-          descriptor.Increment = reader.GetInt64(3);
-          descriptor.MinValue = reader.GetInt64(4);
-          descriptor.MaxValue = reader.GetInt64(5);
-          descriptor.IsCyclic = reader.GetBoolean(6);
-          descriptor.LastValue = reader.GetInt64(7);
+
+          if (!valueReaders.TryGetValue(reader.GetInt32(2), out var valueReader)) {
+            throw new ArgumentOutOfRangeException($"Type of sequence '{reader.GetString(1)}' is not supported.");
```

Comment:
We may have `reader.GetString(1)` as a variable. Now code assumes the current query result row is cached in memory, so it reads column with index 1 again after the column with the index 2 is already read. However, in general, that assumption is not always true. For example `cmd.ExecuteReader(CommandBehavior.SequentialAccess)` in theory must be faster but it doesn't allow stepping back.

Reply:
Sounds reasonable, so I'll make a variable for it.